collectors/beegfsmetaMetric.go

```diff
@@ -0,0 +1,207 @@
+package collectors
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	lp "github.com/ClusterCockpit/cc-metric-collector/internal/ccMetric"
+)
+
+// Struct for the collector-specific JSON config
+type BeegfsMetaCollectorConfig struct {
+	Beegfs            string   `json:"beegfs_path"`
+	ExcludeMetrics    []string `json:"exclude_metrics,omitempty"`
+	ExcludeFilesystem []string `json:"exclude_filesystem"`
+}
+
+type BeegfsMetaCollector struct {
+	metricCollector
+	tags    map[string]string
+	matches map[string]string
+	config  BeegfsMetaCollectorConfig
+	skipFS  map[string]struct{}
+}
+
+func (m *BeegfsMetaCollector) Init(config json.RawMessage) error {
+	// Check if already initialized
+	if m.init {
+		return nil
+	}
+	// Metrics
+	var nodeMdstat_array = [39]string{
+		"sum", "ack", "close", "entInf",
+		"fndOwn", "mkdir", "create", "rddir",
+		"refrEn", "mdsInf", "rmdir", "rmLnk",
+		"mvDirIns", "mvFiIns", "open", "ren",
+		"sChDrct", "sAttr", "sDirPat", "stat",
+		"statfs", "trunc", "symlnk", "unlnk",
+		"lookLI", "statLI", "revalLI", "openLI",
+		"createLI", "hardlnk", "flckAp", "flckEn",
+		"flckRg", "dirparent", "listXA", "getXA",
+		"rmXA", "setXA", "mirror"}
+
+	m.name = "BeegfsMetaCollector"
+	m.setup()
+	// Set default beegfs-ctl binary
+
+	m.config.Beegfs = "/usr/bin/beegfs-ctl"
```

Comment:
This could be a `const`

Reply:
This can be overwritten in the json config so a const would be probably not right.

Reply:
Put a
```
const BEEGFS_CMD "beegfs-ctl"
```
somewhere at the top and assign that one to `m.config.Beefgs` before reading the config JSON. You don't need the absolute path, that's what `LookPath` is for. If the `beegfs-ctl` is not in `$PATH`, the user has to specify it in the config file anyway.